### PR TITLE
[Image] Class to FC

### PR DESCRIPTION
**Pull request by arpitBhalla**
**What kind of change does this PR introduce?**
Convert Class based `Image` Component to `Functional Component`.

**Did you add tests for your changes?**
Yes, I have updated the tests and migrated them to RN-testing Library.

**If relevant, did you update the documentation?**
No changes are required.

**Does this PR introduce a breaking change?**
No, this would not break anything.

**Other information**
GSoC Task of @arpitBhalla.
- [x] Self Review for any kind of error in code.

#### I have tested the PR manually in demo app / Snack
- [x] Android
- [x] iOS
- [x] Web ( Might be solved by using `width` and `height`  )

### CODE REVIEW

File: src/Image/index.tsx [hunk 1]
```diff
@@ -3,4 +3,13 @@ import { Image, ImageProps } from './Image';
...
 export type { ImageProps };
+/**
+ * **React Native Elements Image Component**
+ *
+ * **Documentation** https://reactnativeelements.com/docs/image
+ *
+ * **Preview**
+ *
+ * ![Image](https://raw.githubusercontent.com/arpitBhalla/rne.snippets/next/img/rneImage.png)
+ */
```
Comment (arpitBhalla) [inline src/Image/index.tsx:14]: Screenshot from 2021-06-24 17-07-25

Comment (khushal87) [reply src/Image/index.tsx:16]: https://github.com/react-native-elements/react-native-elements/pull/3112 Please follow the pattern as this, I have discussed today with mentors. Any change related to preview or usage would be done once scripts are written and auto-gen is working as a new feature, not now.

Comment (arpitBhalla) [reply src/Image/index.tsx:16]: Okay 😒

**Comment by pranshuchittora** [comment]
Not able to see images on the web 🤔 
@arpitBhalla can you pls check once.

Screenshot from 2021-07-04 21-51-07

**Comment by arpitBhalla** [comment]
@pranshuchittora It's visible in few components, not sure what's wrong, will check again
image
image
image

**Comment by arpitBhalla** [comment]
@flyingcircle @pranshuchittora Kindly review

**Comment by flyingcircle** [comment]
Screen Shot 2021-07-15 at 3 54 17 PM

There are still parts not showing in web.

**Comment by arpitBhalla** [comment]
> There are still parts not showing in web.

But I only changed Class to FC, 😕

**Comment by arpitBhalla** [comment]
The current `Image` is not working on web as well, might be due to `StyleSheet.absoluteFill` we need to specify `height` and `weight` for Image on web. I guess

**Pull request closed without merging**